File: conditional-execution/python/conditional-Jeremy.py
#!/usr/bin/env python
#
# Introduction to Conditional Execution in python

print '''Pies cost $1 each and depending on much you buy the tax
will change (More pies you buy the less the tax will be).

So, how many pie do you want?'''
tax=0
numPie = input()

# In python blocks are idenifed by indentation and the
# conditional IF doesn't need (), [], or [[]] to separate
# the conditions
if numPie <= 10:
  tax = 0.75
elif numPie <= 25:
  tax = 0.5
elif numPie >= 50:
  tax = 0.25

total = numPie + int(numPie*tax)
print 'Your total for %s pie(s) is $%s' % (numPie, total)

# Switch statements are only in python 3.0 and high, and
# sense I have 2.7 at the moment I don't thing I could
# test a switch so I will just comment it out in this
# section

# switch numPie:
#   case 1:
#     print "That is some weaksauce pie eating."
#   case 10:
#     print "Shovel those things down!"
#   case 25:
#     print "Woh dude, slow down"
#   case 50:
#     print "You don't look so good..."
#   else:
#     print "What too good for the pie?"
# 

File: conditional-execution/perl/conditional-Jeremy.pl
#!/usr/bin/env perl
#
# Introduction to Conditional Execution in perl

use Switch;

@num = (7,2);

# Perl doesn't have >, <, ==, or != so you have to use the
# abbreviated versions
if(@num[0] lt @num[1]) {
  $total = @num[0] + @num[1];
  print "@num[0] + @num[1] = $total\n";
}else {
  $total = @num[0] - @num[1];
  print "@num[0] - @num[1] = $total\n";
}

print "\nQuickly! Coffee, Tea, or Beer.\n";
$drink = "Tea";

# Conditional SWITCH you need to import the Switch to be
# able to use it in perl
switch($drink) {
  case "Coffee" {print "Going to work?\n";}
  case "Tea" {print "Sup hipster!\n";}
  case "Beer" {print "Don't get so drunk.\n";}
}